### PR TITLE
Add blog post about WildFly Operator 0.4.1

### PR DESCRIPTION
**Pull request by jmesnil**
Signed-off-by: Jeff Mesnil <jmesnil@redhat.com>

### CODE REVIEW

**Comment by jmesnil** [comment]
I've set its date to 2020-10-27 (next tuesday)

File: _posts/2020-10-27-wildfly-operator-0.4.1-released.adoc [hunk 1]
```diff
@@ -0,0 +1,355 @@
+---
+layout: post
+title: Release of WildFly Operator 0.4.1
+date: 2020-10-27
+tags: wildfly openshift operator
+synopsis: WildFly Operator now allows seamless upgrades of application deployed on OpenShift.
+author: jmesnil
+---
+
+The WilFly Operator helps deploy and manage WildFly applications on Kubernetes and OpenShift.
```
Comment (rhusar) [inline _posts/2020-10-27-wildfly-operator-0.4.1-released.adoc:10]: s/WilFly/WildFly

File: _posts/2020-10-27-wildfly-operator-0.4.1-released.adoc [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+
+The WildFly operator was using the `applicationImage` field to identify and deploy your application on the container platform.
+This field accepts different types:
+
+* the name of the image: quay.io/jmesnil/wildfly-demo-app:step-1
+* a tag: quay.io/jmesnil/wildfly-demo-app:step-1
+* a digest: quay.io/wildfly-quickstarts/wildfly-operator-quickstart@sha256:0af38bc38be93116b6a1d86a9c78bd14cd527121970899d719baf78e5dc7bfd2
+
+In all these cases, the container platform would pull the image when the application is deployed and uses it for the whole application lifecycle.
+If you wanted to deploy a new image, you had to edit the `applicationImage` manually and specifies the checksum of the new image to deploy.
```
Comment (rhusar) [inline _posts/2020-10-27-wildfly-operator-0.4.1-released.adoc:24]: I think perhaps this was meant to be: s/specifies/specify

File: _posts/2020-10-27-wildfly-operator-0.4.1-released.adoc [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+        String text = "{\"text\":\"" + greetings + ", World!\"}";
+        return text;
+    }
+}
+----
+
+To illustrate that this give access to the full MicroProfile APIs, it is possible to configure the greetings returned by the application by setting the `greetings` config property (which defaults to `"Hello"`).
+It can be configured using an environment variable named `GREETINGS`.
+
+I have built a Docker image of this application and push it to `quay.io/jmesnil/wildfly-demo-app:step-1`.
```
Comment (rhusar) [inline _posts/2020-10-27-wildfly-operator-0.4.1-released.adoc:63]: s/push/pushed

File: _posts/2020-10-27-wildfly-operator-0.4.1-released.adoc [hunk 4]
```diff
@@ -0,0 +1,355 @@
...
+    @GET
+    @Produces({ "application/json" })
+    public String getText() {
+        String text = "{\"text\":\"" + greetings + ", World!\"}";
+        return text;
+    }
+}
+----
+
+To illustrate that this give access to the full MicroProfile APIs, it is possible to configure the greetings returned by the application by setting the `greetings` config property (which defaults to `"Hello"`).
```
Comment (rhusar) [inline _posts/2020-10-27-wildfly-operator-0.4.1-released.adoc:60]: Perhaps this was meant to be s/this give/this gives

File: _posts/2020-10-27-wildfly-operator-0.4.1-released.adoc [hunk 5]
```diff
@@ -0,0 +1,355 @@
...
+Then we tag it with the `latest` tag that will be referenced from our deployments.
+
+[source,bash]
+----
+$ oc tag wildfly-demo-app:step-1  wildfly-demo-app:latest
+----
+
+At this point, we have an `ImageStreamTag` `wildfly-demo-app:latest` that we can use to deploy our application and upgrade it later seamlessly.
+
+We have finally the image inside OpenShift so we can deploy our application it.
```
Comment (rhusar) [inline _posts/2020-10-27-wildfly-operator-0.4.1-released.adoc:218]: 'deploy our application it.' - there is either an extra word here or missing something :-)

File: _posts/2020-10-27-wildfly-operator-0.4.1-released.adoc [hunk 6]
```diff
@@ -0,0 +1,355 @@
...
+To illustrate this, we did a step-by-step demo to understand how and when the upgrade is triggered. However, in normal use, most of these steps are automated and the upgrade becomes really seamless (and do not require user intervention).
+
+There is a lot more that can be achieved by taking full advantage of the OpenShift ecosystem to https://docs.openshift.com/container-platform/4.5/openshift_images/images-understand.html[build and deploy images].
+
+I have built the application image using S2I outside of OpenShift but it is possible to use `BuildConfig` resources to build the image _inside_ OpenShift and automatically tag them with the `latest` imagestream tag.
+Using `BuildConfig` to build the images has two main benefits:
+
+* You can specify hooks to trigger building new images when the code from a remote Git repository is updated.
+* You can also trigger building new images when WildFly S2I images are upgraded. In that case, seamless upgrades apply not only to the application image but also to the 
+
```
Comment (rhusar) [inline _posts/2020-10-27-wildfly-operator-0.4.1-released.adoc:337]: Extra new lines here, would render extraneous paragraph here.

**Review by rhusar — APPROVED** [review]
Cool stuff. Left a couple of comments to fix typos.

File: _posts/2020-10-27-wildfly-operator-0.4.1-released.adoc [hunk 7]
```diff
@@ -0,0 +1,355 @@
...
+$ docker run -p 8080:8080 -e GREETINGS=Bonjour quay.io/jmesnil/wildfly-demo-app:step-1
+...
+$ curl http://localhost:8080
+{"text":"Bonjour, World!"}
+----
+
+[NOTE]
+====
+I created this Docker image from the `step-1` tag of the Git repository. It uses the https://quay.io/repository/wildfly/wildfly-centos7[WildFly 21 S2I image] and the `microprofile-platform` layer so that the image contains only what is needed to run a MicroProfile application.
+The comand I used to create the application image is:
```
Comment (yersan) [inline _posts/2020-10-27-wildfly-operator-0.4.1-released.adoc:78]: s/comand/command

File: _posts/2020-10-27-wildfly-operator-0.4.1-released.adoc [hunk 8]
```diff
@@ -0,0 +1,355 @@
...
+
+[source,java]
+----
+public String getText() {
+    String text = "{\"text\":\"" + (greetings + ", World!").toUpperCase() + "\"}";
+    return text;
+}
+----
+
+This second image is pushed to `quay.io/jmesnil/wildfly-demo-app:step-2` and can can also be run directly from Docker:
```
Comment (yersan) [inline _posts/2020-10-27-wildfly-operator-0.4.1-released.adoc:101]: s/and can can/and can

File: _posts/2020-10-27-wildfly-operator-0.4.1-released.adoc [hunk 9]
```diff
@@ -0,0 +1,355 @@
...
+In a more realistic example, we could have a staging cluster with a `BuildConfig` that would trigger new application images when the code is updated (or when a new Git tag is pushed to a remote repository) or when new WildFly S2I images are released.
+This would automatically trigger a test pipeline to test and verify the new application image.
+
+Once this new application image has been validated (automatically or manually), we can then push this new application image in our production cluster and tag it with `latest` to trigger an upgrade of the application in production.
+
+## Summary
+
+When it is running on OpenShift, the WildFly Operator can leverage its ecosystem around Images to provide seamless upgrades of applications to trigger new deployments
+when anything in the build pipeline (application code or WildFly images) changes.
+This simplifies application maintenance and reduce security risks by automating the upgrades and making sure that the application is always built on top of the latest application code and WildFly images.
```
Comment (yersan) [inline _posts/2020-10-27-wildfly-operator-0.4.1-released.adoc:349]: s/reduce/reduces

**Comment by jmesnil** [comment]
@rhusar @yersan thanks for the review (I'm ashamed I had so many typos).
I've amended the commit with all your suggestions.

**Review by yersan — APPROVED** [review]

**Pull request merged**